src/app.py

```diff
@@ -829,6 +864,22 @@ def publish_datastage(identifier):
                     return jsonify({"error": f"{dataset_uuid} missing contacts or contributors. Must have at least one of each"}), 400
             ingest_helper = IngestFileHelper(app.config)
 
+            # Save a .json file with the metadata information at the top level directory...
+            if dataset_ingest_matadata_dict is not None:
+                json_object = json.dumps(dataset_ingest_matadata_dict['metadata'], indent=4)
+                json_object += '\n'
+                ds_path = ingest_helper.dataset_directory_absolute_path(dataset_data_access_level,
+                                                                        dataset_group_uuid, uuid, False)
+                md_file = os.path.join(ds_path, "metadata.json")
+                logger.info(f"publish_datastage; writing md_file: '{md_file}'; "
+                            f"containing ingest_matadata.metadata: '{json_object}'")
+                try:
+                    with open(md_file, "w") as outfile:
+                        outfile.write(json_object)
+                except IOError as ioe:
```

Comment:
Only handing the `IOError` here is not sufficient. There could be `FileNotFoundError: [Errno 2] No such file or directory`. Maybe use `Exception` to catch all error types?

Reply:
Yes, you are right. If the directory does not exist you an get a FileNotFound Error when writing a file. So, OK.. I'll make that change even though I don't like using broad exception catches in code.